src/Transport/Sending/MessageDispatcher.cs

```diff
@@ -22,7 +22,7 @@ public MessageDispatcher(MessageSenderPool messageSenderPool, string topicName)
         public Task Dispatch(TransportOperations outgoingMessages, TransportTransaction transaction, CancellationToken cancellationToken = default)
         {
             // Assumption: we're not implementing batching as it will be done by ASB client
-            transaction.TryGet<(ServiceBusConnection, string)>(out var receiverConnectionAndPath);
+            transaction.TryGet<(string, string)>(out var receiverConnectionAndPath);
```

Comment:
`ServiceBusConnection` has ensured the key is the connection. `string`, if stands for a connection string, will lead to a new connection which should be avoided.

Reply:
Yeah, I made that change to make the code compile. When we discussed the matter combined with the fact how send-via is done, it seems that the whole MessageSenderPool can only be dependant upon destination. Since the send-via is defined on Client level. So far, the first tests seem to prove it. But we will spend more time on it.